### ISSUE TITLE
Sample input request 

### ISSUE THREAD

**Issue by aysegulbumin** [ctlab/metacherchant#4]
I have run the metacherchant with one read sample in order to try. Even if the input file does not contain character 'B' I keep on getting the same error.  
Would you mind sharing a basic sample input  for which the software runs without any error?
The log  is as follows :

```
08-Jan-18  12:01:58,161  DEBUG   MAIN: Available memory = 2.82 Gb, used = 869 Kb
08-Jan-18  12:01:58,161  DEBUG   MAIN: Available processors = 4
08-Jan-18  12:02:01,719  INFO   MAIN: Running tool environment-finder
08-Jan-18  12:02:01,880  DEBUG  |   BigLong2ShortHashMap: Created 64 small Long2ShortHashMaps
08-Jan-18  12:02:01,880  INFO   |   Loading file f.fastq...
08-Jan-18  12:02:01,898  DEBUG  |   readers-utils: Determined quality format of file f.fastq as Illumina (Phred+64)
08-Jan-18  12:02:01,907  INFO   |   1 reads added
08-Jan-18  12:02:01,907  INFO   |   Loading file r.fastq...
08-Jan-18  12:02:01,907  DEBUG  |   readers-utils: Determined quality format of file r.fastq as Sanger (Phred+33)
08-Jan-18  12:02:01,909  INFO   |   1 reads added
08-Jan-18  12:02:01,909  DEBUG  |   Good/Total sequences count = 2/2 (100.0%)
08-Jan-18  12:02:01,909  DEBUG  |   Good/Total sequences length = 186/186 (100.0%)
08-Jan-18  12:02:01,909  DEBUG  k-mers HM size = 126
08-Jan-18  12:02:01,909  INFO   Hashtable size: 126 kmers
08-Jan-18  12:02:01,956  ERROR  ERROR: Uncaught exception: java.lang.IllegalArgumentException: Incorrect nucleotide char: "B"
java.lang.IllegalArgumentException: Incorrect nucleotide char: "B"
	at ru.ifmo.genetics.dna.DnaTools.fromChar(DnaTools.java:59)
	at ru.ifmo.genetics.dna.DnaQ.<init>(DnaQ.java:27)
	at ru.ifmo.genetics.utils.iterators.DnaQIteratorFromDnaIterator.next(DnaQIteratorFromDnaIterator.java:21)
	at ru.ifmo.genetics.utils.iterators.DnaQIteratorFromDnaIterator.next(DnaQIteratorFromDnaIterator.java:5)
	at ru.ifmo.genetics.io.ReadersUtils.loadDnaQs(ReadersUtils.java:205)
	at tools.EnvironmentFinderMain.loadInput(EnvironmentFinderMain.java:117)
	at tools.EnvironmentFinderMain.runImpl(EnvironmentFinderMain.java:156)
	at ru.ifmo.genetics.utils.tool.Tool.simpleRun(Tool.java:287)
	at ru.ifmo.genetics.utils.tool.Tool.runAsStep(Tool.java:371)
	at ru.ifmo.genetics.utils.tool.Tool.run(Tool.java:451)
	at ru.ifmo.genetics.utils.tool.Tool.mainImpl(Tool.java:607)
	at ru.ifmo.genetics.Runner.run(Runner.java:176)
	at Runner.main(Runner.java:59)
```

**Comment by ropolomx** [comment]
I am getting the same error as well.

**Comment by aysegulbumin** [comment]
Hi,

I have contacted them and we went through their code. The issue is, they assumed that input does not contain any letters other than "A,C,G,T" it is kind of hard coded, which is not necessarily the case for a fastq file. So if your input file contains B,N or other letters then you should adjust it to use this software.

Thanks.

**Comment by ropolomx** [comment]
Thanks for your reply, @aysegulbumin. This error also occurs when I use FASTA files (not FASTQ) and none of the sequences have a B or a N character. This is with the latest version from `bioconda` (0.1.0). For more information, these are the last lines of the log:
```
24-Jul-18  12:51:58  Splitting 99 into tokens
24-Jul-18  12:51:58  Splitting 98 into tokens
24-Jul-18  12:51:58  Splitting 95 into tokens
24-Jul-18  12:51:58  Splitting 99 into tokens
24-Jul-18  12:51:58  |   936'852 reads added
24-Jul-18  12:51:58  Hashtable size: 60184369 kmers
24-Jul-18  12:51:58  ERROR: Uncaught exception: java.lang.IllegalArgumentException: Incorrect nucleotide char: "B"
java.lang.IllegalArgumentException: Incorrect nucleotide char: "B"
	at ru.ifmo.genetics.dna.DnaTools.fromChar(DnaTools.java:59)
	at ru.ifmo.genetics.dna.DnaQ.<init>(DnaQ.java:27)
	at ru.ifmo.genetics.utils.iterators.DnaQIteratorFromDnaIterator.next(DnaQIteratorFromDnaIterator.java:21)
	at ru.ifmo.genetics.utils.iterators.DnaQIteratorFromDnaIterator.next(DnaQIteratorFromDnaIterator.java:5)
	at ru.ifmo.genetics.io.ReadersUtils.loadDnaQs(ReadersUtils.java:205)
	at tools.EnvironmentFinderMain.loadInput(EnvironmentFinderMain.java:117)
	at tools.EnvironmentFinderMain.runImpl(EnvironmentFinderMain.java:156)
	at ru.ifmo.genetics.utils.tool.Tool.simpleRun(Tool.java:287)
	at ru.ifmo.genetics.utils.tool.Tool.runAsStep(Tool.java:371)
	at ru.ifmo.genetics.utils.tool.Tool.run(Tool.java:451)
	at ru.ifmo.genetics.utils.tool.Tool.mainImpl(Tool.java:607)
	at ru.ifmo.genetics.Runner.run(Runner.java:176)
	at Runner.main(Runner.java:59)
```

**Comment by ropolomx** [comment]
I think I found what is going on... The MEGAREs database, which I am using as sequence input, does contain a few sequences with degenerate bases ("Y", "B", etc.). I think this is what is causing the problems. I will test again after resolving the degenerate bases. Are you using MetaCherchant with the MEGARes database too?

**Comment by aysegulbumin** [comment]
Yes I used MEGARes database as well. 
After replacing the other letters with any of A,C,G,Ts I was able to run.
Hopefully this will be helpful.

**Comment by ivartb** [comment]
A fix was added, so now all degenerate bases can be used in input sequences. Solved by #7
